CMakeLists.txt

```diff
@@ -23,7 +22,45 @@ project(${SDK_NAME}
     LANGUAGES CXX
     VERSION ${OPENDAQ_PACKAGE_VERSION}
 )
-add_compile_definitions(OPENDAQ_PACKAGE_VERSION="${OPENDAQ_PACKAGE_VERSION}")
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake
+)
+
+get_filename_component(ROOT_DIR ${CMAKE_SOURCE_DIR} REALPATH)
+get_filename_component(PROJ_DIR ${PROJECT_SOURCE_DIR} REALPATH)
+
+if (NOT ${PROJECT_SOURCE_DIR} STREQUAL ${ROOT_DIR})
+    set(BUILDING_AS_SUBMODULE ON)
+    message(STATUS "Building as submodule")
+else()
+    message(STATUS "Building standalone")
+    set_property(GLOBAL PROPERTY PREDEFINED_TARGETS_FOLDER ".CMakePredefinedTargets")
+endif()
+
+if (NOT BUILDING_AS_SUBMODULE)
+    find_package(GitInfo)
+    if (COMMAND GIT_WC_INFO)
+        GIT_WC_INFO("${CMAKE_CURRENT_SOURCE_DIR}" "OPENDAQ")
+		if (DEFINED CI_GIT_BRANCH AND NOT ${CI_GIT_BRANCH} STREQUAL "")
+		    STRING(REPLACE "refs/heads/" "" OPENDAQ_WC_BRANCH_NAME ${CI_GIT_BRANCH})
+		endif()
+		
+        message(STATUS "Git branch: ${OPENDAQ_WC_BRANCH_NAME} | ${OPENDAQ_WC_REVISION_HASH}")
+	    if (${OPENDAQ_WC_BRANCH_NAME} STREQUAL "main")
```

Comment:
What if openDAQ is built from another branch? I would except that we add a branch name in this case?

Reply:
True, we could simply leave the hash in unless the branch matches `release/**`, would likely be a better solution.